python/docs/source/examples/time_centric.ipynb

```diff
@@ -0,0 +1,339 @@
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "6b5f2725",
+   "metadata": {},
+   "source": [
+    "\n",
+    "Beyond querying for events, Kaskada has a powerful syntax for defining\n",
+    "calculations on events, temporally across history.\n",
+    "\n",
+    "The six calculations discussed at the top of this demo notebook are below.\n",
+    "\n",
+    "(Note that some functions return `NaN` if no events for that user have occurred\n",
+    "within the time window.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3ad6d596",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "Purchases = Events.filter(Events.col(\"event_name\").eq(\"purchase\"))\n",
+    "\n",
+    "Features = kd.record({\n",
+    "    \"event_count_total\": Events.count(),\n",
+    "    #\"event_count_hourly\": Events.count(window=Hourly()),\n",
+    "    \"purchases_total_count\": Purchases.count(),\n",
+    "    #\"purchases_today\": Purchases.count(window=Since(Daily()),\n",
+    "    #\"revenue_today\": Events.col(\"revenue\").sum(window=Since(Daily())),\n",
+    "    \"revenue_total\": Events.col(\"revenue\").sum(),\n",
+    "})\n",
+    "Features.preview()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "1c315938",
+   "metadata": {},
+   "source": [
+    "#### At Any Time\n",
+    "\n",
+    "A key feature of Kaskada's time-centric design is the ability to query for\n",
+    "calculation values at any point in time. Traditional query languages (e.g. SQL)\n",
+    "can only return data that already exists---if we want to return a row of\n",
+    "computed/aggregated data, we have to compute the row first, then return it. As a\n",
+    "specific example, suppose we have SQL queries that produce daily aggregations\n",
+    "over event data, and now we want to have the same aggregations on an hourly\n",
+    "basis. In SQL, we would need to write new queries for hourly aggregations; the\n",
+    "queries would look very similar to the daily ones, but they would still be\n",
+    "different queries.\n",
+    "\n",
+    "With Kaskada, we can define the calculations once, and then specify the points\n",
+    "in time at which we want to know the calculation values when we query them.\n",
+    "\n",
+    "In the examples so far, we have used `preview()` to get a DataFrame containing\n",
+    "some of the rows from the Timestreams we've defined. By default, this produces\n",
+    "a _history_ containing all the times the result changed. This is useful for\n",
+    "using past values to create training examples.\n",
+    "\n",
+    "We can also execute the query for the values at a specific point in time."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "082e174d",
+   "metadata": {
+    "tags": [
+     "hide-output"
+    ]
+   },
+   "source": [
+    "```\n",
+    "Features.preview(at=\"2022-01-01 22:00\")\n",
```

Comment:
Not a cell yet since we don't have "final values at time" hooked up. But, it seems like this would be an easy thing to add to flesh out this notebook.